src/pages/home/ReportScreen.js

```diff
@@ -350,6 +351,18 @@ function ReportScreen({
         [report, isLoading, shouldHideReport, isDefaultReport, isOptimisticDelete, userLeavingStatus],
     );
 
+    useEffect(() => {
+        if (isFocused) {
+            return;
+        }
+
+        if (ReportUtils.isThread(report) && report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN) {
+            Report.leaveRoom(report.reportID, false);
+        }
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Agree we don't want to run this whenever any property on the report changes. However, I think we probably want to make sure that the correct report data is used and want to avoid suppressing the lint rule.

So maybe we can workaround the lint rule with something like this:

```
const isThread = ReportUtils.isThread(report);
const isNotificationPreferenceHidden = report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
const reportID = report.reportID;

// When the report screen is unmounted or no longer in focus, and the user has no comments on that thread, call LeaveRoom
useEffect(() => {
    if (isFocused) {
        return;
    }

    if (!isThread || !isNotificationPreferenceHidden) {
        return;
    }

    Report.leaveRoom(report.reportID, false);
}, [isFocused, isThread, isNotificationPreferenceHidden]);
```